### PR TITLE
Da User Story 18

### PR DESCRIPTION
**Pull request by d-atkins**
## Description
Completes User Story #18

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which maintains existing functionality)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Notes
Adds statistics to /items: top 5 most/least popular (most/least ordered) items
  - adds class method `self.popular(limit, order)` to `Item` model
    - limit - integer specifying number of records to return
    - order - string of either "desc" (for most popular) or "asc" (for least popular)

## RSpec Results
```
Finished in 5.52 seconds (files took 2.21 seconds to load)
124 examples, 0 failures

Coverage report generated for RSpec - 1708 / 1708 LOC (100.0%) covered.
```

### CODE REVIEW

File: spec/models/item_spec.rb [hunk 1]
```diff
@@ -62,7 +62,51 @@
...
+      100.times { create(:random_item_order, item: super_item, price: item1.price, quantity: 15) }
+      8.times { create(:random_item_order, item: item5, price: item1.price, quantity: 5) }
+      3.times { create(:random_item_order, item: item6, price: item1.price, quantity: 5) }
+      1.times { create(:random_item_order, item: item7, price: item1.price, quantity: 5) }
+
+      expect(Item.popular(5, "desc")).to eq([super_item, item1, item2, item3, item4])
+      expect(Item.popular(5, "asc")).to eq([item7, item6, item5, item4, item3])
     end
+
+    # it 'least_popular' do
```
Comment (kathleen-crowe) [inline spec/models/item_spec.rb]: we should delete this code before merging otherwise everything is good

**Review by kathleen-crowe — CHANGES_REQUESTED** [review]

**Review by kathleen-crowe — APPROVED** [review]

File: app/controllers/admin/merchants_controller.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@ def index
 
   def update
     merchant = Merchant.find(params[:id])
-    merchant.toggle!(:active)
+    merchant.toggle!(:active?)
```
Comment (iEv0lv3) [inline app/controllers/admin/merchants_controller.rb:8]: This fixes the failing test, nice!

File: app/models/item.rb [hunk 3]
```diff
@@ -12,6 +12,13 @@ class Item <ApplicationRecord
   validates_inclusion_of :active?, :in => [true, false]
   validates_numericality_of :price, greater_than: 0
 
+  def self.popular(limit, order)
+     Item.select("items.*, SUM(quantity) AS total")
+       .joins(:item_orders)
+       .group(:id)
+       .order("total #{order}")
+       .limit(limit)
+  end
```
Comment (iEv0lv3) [inline app/models/item.rb:21]: Hella query!

File: app/views/items/index.html.erb [hunk 4]
```diff
@@ -23,3 +31,11 @@
...
     <% end %>
 </section>
+<section id="least_popular"><h1>Least Popular Items</h1>
+  <% @items.popular(5, 'asc').each.with_index do |item, index| %>
+    <section id="bottom-<%= index + 1 %>">
+      <p><%= link_to item.name, "/items/#{item.id}" %></p>
+      <p>Total amount ordered: <%= item.quantity_bought %></p>
+    </section>
+  <% end %>
+</section>
```
Comment (iEv0lv3) [inline app/views/items/index.html.erb:41]: Solid conditional logic here.

File: spec/factories/item_order.rb [hunk 5]
```diff
@@ -1,5 +1,5 @@
 FactoryBot.define do
-  factory :random_item_order, class: Item do
+  factory :random_item_order, class: ItemOrder do
```
Comment (iEv0lv3) [inline spec/factories/item_order.rb:2]: Great catch here to apply the correct class.

**Review by iEv0lv3 — APPROVED** [review]
* Great PR
* Love how comprehensive the tests are
* Functionality is as expected
* Fixes are great
* Test coverage is 100%

**Pull request merged**